docs/spec.md

```diff
@@ -6247,6 +6277,27 @@ for `is_quantized(x) and quantization_dimension(x) is not None`.
 * `is_per_tensor_quantized(x: Value | Placeholder | Type) -> Value` is a
 shortcut for `is_quantized(x) and quantization_dimension(x) is None`.
 
+* `is_promotable(x: Type, y: Type) -> bool` checks if type `x` can be promoted
+to type `y`.  When `x` and `y` are `QuantizedTensorElementType`s, the promotion
+is applied only to the `storage_type`. This specific version of promotion is
+currently used in context of reduction computation (refer to
+[RFC](https://github.com/openxla/stablehlo/pull/1664) for more details).
+
+```python
+def is_promotable(x: Type, y: Type) -> Value:
```

Comment:
>> Does "promotable" have any implications on bitwidth?

In the current form there is no implication and it does allow conversion in both ways (i32 -> i16 and i16 -> i32). 

Per my offline chat with @loganchien, the output type can be anything to which the accumulation result needs to be converted. So all of `f32->f32` (no conversion), `f32 -> f16`, `f16 -> f32` are allowed.


>> I believe the existing rules permit this, not sure if we want this.
There does not seem like a current use case with the demotion without overflowing the accumulation result (confirmed with @loganchien). IMO,  In its current form it provides generality to the user get the output in any precision they want.

With that I can think of two options:
1. Keep the generality as is. With that, I agree that `is_promotable` might be ambiguous. How about we use something like `is_implicitly_convertable`. There I would like to quality `_convertable` somehow to highlight that this is not a general conversion, but for a specific use-case. What do you think?

2. Add additional check to allow `bitwidth(source) <= bitwidth(destination)` 

I am biased towards the general approach with less constraints, but I would like to know your opinion.

Reply:
I like general if there are use cases for being general. Rationale: Everything we support in the IR has implications for what backends must support.

With that said - Do we think it is trivial overhead for backends to support demotion if we are already supporting promotion? If so then probably OK to be generic. But if it would require more specialization for each of these ops, then perhaps we reconsider and restrict to use cases.

Reply:
Hi Kevin, thanks for the feedback!

The specific question bought up in the above discussion is: should we promote demotion as well? It seems that there aren't any use-cases for that. For example, consider the following cases of implicit conversion for reduction operations. The examples are shown using integer type but can be generalized to other types as well.
 
| input element type | accumulator element type  | result element type | use-cases |
|:--:|:--:|:--:|:--|
| i8| i8 |i8| Reduction using min/max|
| i8| i32 |i32| Reduction using average/sum|
| i8| i32| i8 | Demotion from accumulation type to result type. (No use-case)|
| i8| i32| i64 | Extra promotion from accumulation type to result type. (No use-case) |
| i32 | i8 | i8 | Demotion from input value type to accumulation type. (No use case) |

With that in mind, we propose to develop the specification based on the scenario we have the use-case for (just for promotion). 
Thanks @loganchien for you input on this.  

@loganchien @GleasonK @paulinesho If the above proposal looks good to you, can you please review the PR. The only changes from the last time are
1. The definition of `is_promotable` now only support the promotion cases with `bit-width(element_type(input)) <= bit-width(element_type(accumulator))`.
2. For all the ops supported in the PR: `element_type(result) == element_type(accumulator)`.

Reply:
SGTM.

Reply:
LGTM